posthog/hogql_queries/events_query_runner.py

```diff
@@ -263,5 +263,12 @@ def apply_dashboard_filters(self, dashboard_filter: DashboardFilter):
         if dashboard_filter.properties:
             self.query.properties = (self.query.properties or []) + dashboard_filter.properties
 
+    def columns(self, query_columns: list | None) -> list[str]:
+        columns = query_columns or []
+        return [
+            col if col == "*" else columns[idx] if len(columns) > idx else col
+            for idx, col in enumerate(self.select_input_raw())
+        ]
```

Comment:
Needed to add this to retain the behaviour in `posthog/api/test/test_query.py`

https://github.com/PostHog/posthog/blob/088a1324baeb78543e0e1a71d8a55fe97790d0e3/posthog/api/test/test_query.py#L105-L107

If I just had `columns=query_result.columns` the test would need to be changed to something like:
```
self.assertEqual(response["columns"], ["tuple(<ALL THE COLUMNS>)", "event"])
```

Happy to go with either approach but will leave it up to @PostHog/team-product-analytics to decide because I'm not sure what the correct result is here